### PR TITLE
Feature/ATG-15 Option component

### PR DESCRIPTION
**Pull request by IraCherginyshka**
save customize controls in localstorage
update logic in controls component & state-controls component

### CODE REVIEW

File: src/app/components/game-control-option/game-control-option.component.ts [hunk 1]
```diff
@@ -1,8 +1,61 @@
...
+export class GameControlOptionComponent implements OnInit {
+  public controls: ControlsStateObject;
+  public readonly controlsEnum = ControlsEnum;
+  public readonly defaultSettings = DefaultSettings;
+  private isSetMode = false;
+  private activeField: string;
+
+  ngOnInit(): void {
+    if (localStorage.getItem('controls')) {
+      const savedControls = JSON.parse(localStorage.getItem('controls'));
```
Comment (oleksandrstarov) [inline src/app/components/game-control-option/game-control-option.component.ts]: I suggest to use const savedControls = JSON.parse(localStorage.getItem('controls'));

if (savedControls) ...
else ...

File: src/app/components/game-controls/game-controls.component.ts [hunk 2]
```diff
@@ -28,6 +32,17 @@ export class GameControlsComponent implements OnInit, OnDestroy {
     this.subscriptionLost = this.gameService.onLostGame().subscribe(() => {
       this.isLostGame = true;
     });
+
+    if (localStorage.getItem('controls')) {
+      this.controls = JSON.parse(localStorage.getItem('controls'));
+    } else {
```
Comment (oleksandrstarov) [inline src/app/components/game-controls/game-controls.component.ts]: Same as previous comment

File: src/app/components/game-state-controls/game-state-controls.component.ts [hunk 3]
```diff
@@ -23,6 +27,16 @@ export class GameStateControlsComponent implements OnInit, OnDestroy {
     this.subscriptionState = this.gameService.getGameState().subscribe((action: GameState) => {
       this.isPlaying = action !== GameState.PAUSE;
     });
+
+    if (localStorage.getItem('controls')) {
+      this.controls = JSON.parse(localStorage.getItem('controls'));
+    } else {
```
Comment (oleksandrstarov) [inline src/app/components/game-state-controls/game-state-controls.component.ts]: Same as above

File: src/app/interfaces/controlsState.interface.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { DefaultSettings } from '../enums/default-settings.enum';
+
+export interface ControlsStateObject {
+  reset?: string | DefaultSettings;
+  pause?: string | DefaultSettings;
+  play?: string | DefaultSettings;
+  rotate?: string | DefaultSettings;
+  left?: string | DefaultSettings;
+  drop?: string | DefaultSettings;
+  right?: string | DefaultSettings;
```
Comment (oleksandrstarov) [inline src/app/interfaces/controlsState.interface.ts]: Not sure why do we need to use optional fields? cant find any particular place. If there requirements for this in single place - we can use Partial<ControlsStateObject> https://www.typescriptlang.org/docs/handbook/utility-types.html#partialt

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

File: src/app/components/game-control-option/game-control-option.component.ts (hunk 1 above)
Comment (IraCherginyshka) [reply src/app/components/game-control-option/game-control-option.component.ts]: Changed

File: src/app/components/game-controls/game-controls.component.ts (hunk 2 above)
Comment (IraCherginyshka) [reply src/app/components/game-controls/game-controls.component.ts]: Changed

File: src/app/components/game-state-controls/game-state-controls.component.ts (hunk 3 above)
Comment (IraCherginyshka) [reply src/app/components/game-state-controls/game-state-controls.component.ts]: Changed

File: src/app/interfaces/controlsState.interface.ts (hunk 4 above)
Comment (IraCherginyshka) [reply src/app/interfaces/controlsState.interface.ts]: Done

**Review by oleksandrstarov — APPROVED** [review]

**Pull request merged**